staging/src/k8s.io/apiserver/pkg/server/egressselector/egress_selector.go

```diff
@@ -203,7 +203,8 @@ func (u *udsGRPCConnector) connect() (proxier, error) {
 		return c, err
 	})
 
-	tunnel, err := client.CreateSingleUseGrpcTunnel(udsName, dialOption, grpc.WithInsecure())
+	ctx := context.TODO()
+	tunnel, err := client.CreateSingleUseGrpcTunnel(ctx, udsName, dialOption, grpc.WithInsecure())
```

Comment:
context.TODO() here is not good... it means the timeout of the dial is not connected to the timeout of the triggering request... plumbing the request context into connect() would let us connect those

Reply:
This is intended to be a placeholder while we determine what actual context should be plumbed through

Reply:
@cheftako can elaborate

Reply:
ok. @cheftako will follow-up to wire this up once he confirms the lifetime of the request context matches

as is, this PR gets us context timeouts on the dial and connection to the backend, which is the most critical